receiver/statsdreceiver/transport/unix_server.go

```diff
@@ -0,0 +1,133 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package transport // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver/transport"
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"sync"
+	"syscall"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.uber.org/multierr"
+)
+
+type unixServer struct {
+	listener    net.Listener
+	reporter    Reporter
+	connections map[net.Conn]struct{}
+	wg          sync.WaitGroup
+	lock        *sync.Mutex
+	socket      string
+}
+
+var _ (Server) = (*unixServer)(nil)
+
+// NewUnixServer creates a transport.Server listening from a Unix Socket.
+func NewUnixServer(socket string) (Server, error) {
+	err := os.Remove(socket)
```

Comment:
why removing the socket?

Reply:
In case a collector crashed previously, we use a new socket for the new connections to not fail with "bind: address already in use"

In case there are two collectors running because of a misconfiguration, new clients connect to the new socket. Clients connected to the previous instances can still connect to the old socket if they are already connected.